### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by EksdeeeX**
- Added feature implementation of 'unmark' and design considerations.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -176,6 +176,29 @@ The mark feature mechanism is implemented by having the Event class keep track o
...
+- A valid `index` will be taken in as a parameter to facilitate unmarking of the correct event.
+- The unmark feature builds on the existing implementation of the 'Mark' class.
+- Allows for undo of accidental marking by setting the 'isDone' boolean attribute of the 'Mark' class to false to signify an undone event.
+- Unmarking of an undone event would not result in error or exceptions as this action does not reduce correctness of presented information.
+
+#### Design considerations:  
+
+* **Alternative 1 (current choice): Calls on `unmark` method in 'Mark' class to set the tracked boolean variable 'isDone' to false** 
+  * Pros: Easier to implement.
+  * Cons: Results in higher coupling as it is editing a vairable in an event instance, may lead to errors in runtime and testing.
```
Comment (ExtraShotLatte) [inline docs/DeveloperGuide.md:196]: 'variable' instead of 'vairable' 😅

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -176,6 +176,29 @@ The mark feature mechanism is implemented by having the Event class keep track o
...
+### Unmark feature
+
+The `unmark` feature allows for unmarking of the 'Mark' attribute in events should there be an accidental marking of an event.
+
+#### Implementation
+
+The `unmark` feature is implemented with the following considerations:
+
+- A valid `index` will be taken in as a parameter to facilitate unmarking of the correct event.
+- The unmark feature builds on the existing implementation of the 'Mark' class.
```
Comment (ExtraShotLatte) [inline docs/DeveloperGuide.md:188]: Is 'build alongside' or equivalent better? 'builds on' to me sounds like there are dependencies involved.

**Review by ExtraShotLatte — CHANGES_REQUESTED** [review]
LGTM! Just some spelling error and and word choices that I'm unsure about! 👍

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (EksdeeeX) [reply docs/DeveloperGuide.md:215]: Wow, good catch. amended 👍

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (EksdeeeX) [reply docs/DeveloperGuide.md:207]: yup sounds good, amended 👍

**Review by ExtraShotLatte — APPROVED** [review]
LGTM! Thanks for the quick amendments!

**Pull request merged**